src/xas_tdp_correction.F

```diff
@@ -0,0 +1,1870 @@
+        dbcsr_copy, dbcsr_create, dbcsr_distribution_get, dbcsr_distribution_new, &
+        dbcsr_distribution_release, dbcsr_distribution_type, dbcsr_get_info, dbcsr_p_type, &
+        dbcsr_release, dbcsr_scalar, dbcsr_type
+   USE dbcsr_tensor_api,                ONLY: &
+        dbcsr_t_contract, dbcsr_t_copy, dbcsr_t_copy_matrix_to_tensor, dbcsr_t_create, &
+        dbcsr_t_default_distvec, dbcsr_t_destroy, dbcsr_t_distribution_destroy, &
+        dbcsr_t_distribution_new, dbcsr_t_distribution_type, dbcsr_t_finalize, dbcsr_t_get_block, &
+        dbcsr_t_get_info, dbcsr_t_iterator_blocks_left, dbcsr_t_iterator_next_block, &
+        dbcsr_t_iterator_start, dbcsr_t_iterator_stop, dbcsr_t_iterator_type, &
+        dbcsr_t_pgrid_create, dbcsr_t_pgrid_destroy, dbcsr_t_pgrid_type, dbcsr_t_put_block, &
+        dbcsr_t_type
+   USE hfx_admm_utils,                  ONLY: create_admm_xc_section
+   USE input_section_types,             ONLY: section_vals_create,&
+                                              section_vals_get_subs_vals,&
+                                              section_vals_release,&
+                                              section_vals_retain,&
+                                              section_vals_set_subs_vals,&
+                                              section_vals_type
+   USE kinds,                           ONLY: dp
+   USE machine,                         ONLY: m_flush
+   USE message_passing,                 ONLY: mp_sum
+   USE physcon,                         ONLY: evolt
+   USE qs_environment_types,            ONLY: get_qs_env,&
+                                              qs_environment_type
+   USE qs_ks_methods,                   ONLY: qs_ks_build_kohn_sham_matrix
+   USE qs_mo_types,                     ONLY: deallocate_mo_set,&
+                                              duplicate_mo_set,&
+                                              get_mo_set,&
+                                              mo_set_p_type,&
+                                              reassign_allocated_mos
+   USE util,                            ONLY: get_limit
+   USE xas_tdp_kernel,                  ONLY: contract2_AO_to_doMO,&
+                                              ri_all_blocks_mm
+   USE xas_tdp_types,                   ONLY: donor_state_type,&
+                                              xas_tdp_control_type,&
+                                              xas_tdp_env_type
+
+!$ USE OMP_LIB, ONLY: omp_get_max_threads, omp_get_thread_num
+#include "./base/base_uses.f90"
+
+   IMPLICIT NONE
+   PRIVATE
+
+   CHARACTER(len=*), PARAMETER, PRIVATE :: moduleN = 'xas_tdp_correction'
+
+   PUBLIC :: gw2x_shift, get_soc_splitting
+
+CONTAINS
+
+! **************************************************************************************************
+!> \brief First implementation of the very legit GW2X correction of the Kohn-Sham IP from
+!>        https://onlinelibrary.wiley.com/doi/pdf/10.1002/qua.1543
```

Comment:
maybe do a bibliography entry for this one?

Reply:
Yes you are right. Also, the text could be somewhat updated.